src/AdminConnection.ts

```diff
@@ -1,1213 +1,1218 @@
+	CompactAdapterInfo,
+	CompactHost,
+	CompactInstalledInfo,
+	CompactInstanceInfo,
+	CompactRepository,
+	Logfile,
+} from "./SocketEvents";
+import {
+	getObjectViewResultToArray,
+	normalizeHostId,
+	objectIdToHostname,
+} from "./tools";
+
+interface Certificate {
+	name: string;
+	type: "public" | "private" | "chained";
+}
+
+function parseCertificate(name: string, cert: string): Certificate | undefined {
+	if (!cert) return;
+
+	let type: Certificate["type"];
+	// If it is filename, it could be everything
+	if (
+		cert.length < 700 &&
+		(cert.indexOf("/") !== -1 || cert.indexOf("\\") !== -1)
+	) {
+		if (name.toLowerCase().includes("private")) {
+			type = "private";
+		} else if (cert.toLowerCase().includes("private")) {
+			type = "private";
+		} else if (name.toLowerCase().includes("public")) {
+			type = "public";
+		} else if (cert.toLowerCase().includes("public")) {
+			type = "public";
+		} else {
+			// TODO: is this correct?
+			return undefined;
+		}
+	} else {
+		type =
+			cert.substring(0, "-----BEGIN RSA PRIVATE KEY".length) ===
+				"-----BEGIN RSA PRIVATE KEY" ||
+			cert.substring(0, "-----BEGIN PRIVATE KEY".length) ===
+				"-----BEGIN PRIVATE KEY"
+				? "private"
+				: "public";
+
+		if (type === "public") {
+			const m = cert.split("-----END CERTIFICATE-----");
+			if (
+				m.filter((t) => t.replace(/\r\n|\r|\n/, "").trim()).length > 1
+			) {
+				type = "chained";
+			}
+		}
+	}
+	return { name, type };
+}
+
+interface IPAddress {
+	name: string;
+	address: string;
+	family: "ipv4" | "ipv6";
+}
+
+interface IPAddresses {
+	IPs4: IPAddress[];
+	IPs6: IPAddress[];
+}
+
+function parseIPAddresses(host: ioBroker.HostObject): IPAddresses {
+	const IPs4: IPAddress[] = [
+		{
+			name: "[IPv4] 0.0.0.0 - Listen on all IPs",
+			address: "0.0.0.0",
+			family: "ipv4",
+		},
+	];
+	const IPs6: IPAddress[] = [
+		{
+			name: "[IPv6] :: - Listen on all IPs",
+			address: "::",
+			family: "ipv6",
+		},
+	];
+	if (host.native?.hardware?.networkInterfaces) {
+		for (const [eth, iface] of Object.entries(
+			host.native.hardware.networkInterfaces,
+		)) {
+			if (!iface) continue;
+
+			for (const ip of iface) {
+				if (ip.family !== "IPv6") {
+					IPs4.push({
+						name: `[${ip.family}] ${ip.address} - ${eth}`,
+						address: ip.address,
+						family: "ipv4",
+					});
+				} else {
+					IPs6.push({
+						name: `[${ip.family}] ${ip.address} - ${eth}`,
+						address: ip.address,
+						family: "ipv6",
+					});
+				}
+			}
+		}
+	}
+	return { IPs4, IPs6 };
+}
+export class AdminConnection extends Connection<
+	AdminListenEvents,
+	AdminEmitEvents
+> {
+	constructor(props: ConnectionProps) {
+		super(props);
+	}
+
+	/**
+	 * Get the stored certificates.
+	 * @param update Force update.
+	 */
+	getCertificates(update?: boolean): Promise<Certificate[]> {
+		return this.request({
+			cacheKey: "cert",
+			forceUpdate: update,
+			// TODO: check if this should time out
+			commandTimeout: false,
+			requireAdmin: true,
```

Comment:
(Almost) all methods in this class require admin, it would be good to overload the method in this class and have `requireAdmin` always set to `true`.